lib/lightning_web/live/workflow_live/form.ex

```diff
@@ -0,0 +1,145 @@
+              <button
+                disabled={!@isButtonDisabled}
+                type="submit"
+                class=" justify-center rounded-md bg-primary-600 disabled:bg-primary-300 px-6 py-3 text-sm font-semibold text-white shadow-sm hover:bg-primary-500 disabled:outline-0 focus:outline-2 focus:outline-indigo-600 focus:outline-offset-2 active:outlin-2 active:outline-indigo-600 active:outline-offset-2"
+              >
+                Create Workflow
+                <%= if !@isButtonDisabled do %>
+                  <span class="invisible group-hover:visible w-36 py-1 px-3 bg-[#030712] absolute  rounded-md -translate-y-16 -translate-x-32">
+                    A workflow name is required
+                  </span>
+                <% end %>
+              </button>
+            </span>
+          </div>
+        </.modal_footer>
+      </.form>
+    </div>
+    """
+  end
+
+  @impl true
+  def update(assigns, socket) do
+    changeset = validate_workflow_name(@form_fields)
+
+    socket =
+      socket
+      |> assign(:form, to_form(changeset, as: :input_form))
+      |> assign(:project_id, assigns.id)
+      |> assign(:isButtonDisabled, changeset.valid?)
+
+    {:ok, socket}
+  end
+
+  @impl true
+  def handle_event("validate", %{"workflow_name" => workflow_name}, socket) do
+    changeset = validate_workflow(workflow_name, socket)
+
+    socket =
+      socket
+      |> assign(:isButtonDisabled, changeset.valid?)
+
+    {:noreply, assign(socket, form: to_form(changeset, as: :input_form))}
+  end
+
+  @impl true
+  def handle_event(
+        "create_work_flow",
+        %{"workflow_name" => workflow_name},
+        socket
+      ) do
+    changeset = validate_workflow(workflow_name, socket)
+
+    if changeset.valid? do
+      navigate_to_new_workflow(socket, workflow_name)
+    else
+      {:noreply, update_form(socket, changeset)}
+    end
+  end
+
+  defp validate_workflow(workflow_name, socket) do
+    validate_workflow_name(@form_fields, %{
+      name: workflow_name,
+      project_id: socket.assigns.project_id
+    })
+    |> Map.put(:action, :validate)
+  end
+
+  defp navigate_to_new_workflow(socket, workflow_name) do
+    {:noreply,
+     push_navigate(socket,
+       to:
+         ~p"/projects/#{socket.assigns.project_id}/w/new?#{%{name: workflow_name}}"
+     )}
+  end
+
+  defp update_form(socket, changeset) do
+    assign(socket, :form, to_form(changeset, as: :input_form))
+  end
+
+  defp changeset(workflow, attrs) do
+    {workflow, @types}
+    |> cast(attrs, Map.keys(@types))
+    |> validate_required([:name])
+    |> validate()
+  end
+
+  defp validate(changeset) do
+    workflow_name = get_field(changeset, :name)
+    project_id = get_field(changeset, :project_id)
+
+    if workflow_name && project_id do
+      case Workflows.workflow_exists?(project_id, workflow_name) do
+        true ->
+          add_error(changeset, :name, "Workflow name already been used")
+
+        false ->
+          changeset
```

Comment:
On Erlang coding this is okay but on Elixir would use the `if` for boolean expression or a `cond` to simplify two cases of returning a `changeset`.